src/main/preprocessing/preprocessing.py

```diff
@@ -34,11 +37,23 @@ def __merge_ati_files(ati_files: List[str]) -> pd.DataFrame:
 
 
 def is_test_mode(ct_df: pd.DataFrame) -> bool:
+    # The old version of the data does not contains the test mode column. We should handle this case correctly anyway.
+    if CODE_TRACKER_COLUMN.TEST_MODE.value not in ct_df.columns:
+        return False
     return ct_df[CODE_TRACKER_COLUMN.TEST_MODE.value].values[0] == TEST_MODE.ON.value
 
 
-# The function returns True if the code tracker file was created and False otherwise
 def __handle_ct_files(ct_files: List[str], output_task_path: str) -> bool:
+    """
+    The function returns True if new code tracker file was created and False otherwise
+    We should choose the last state of the code tracker files for the task or all last states and create a new file
+    where we union them. The student can submit the solution several times, while the history of the codetracker file
+    is not erased. In this way, we only need to select the final file with the entire history. On the other hand,
+    if the file was full, then it will be sent additionally and new files will contain a new history.
+    In this case, it is necessary to find the last states of all files with a unique history, combine according to
+    timestamps and write to a new final file.
```

Comment:
hmmm can we combine files that are full?

Reply:
I added todo: `# TODO: handle the case with additional files`

I think we should do it in the next PR. But if you want I can do it now

Reply:
Let's leave it till the next PR